File: myimages_ui/src/services/GalleryService.js
import http from "../http-common"; 

class GalleryService {
  getAllGallerys(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/gallery/gallerys`, searchDTO);
  }
 

  get(galleryId) {
    return this.getRequest(`/gallery/${galleryId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/gallery?field=${matchData}`, null);
  }

  addGallery(data) {
    return http.post("/gallery/addGallery", data);
  }

  update(data) {
  	return http.post("/gallery/updateGallery", data);
  }
  
  uploadImage(data,galleryId) {
  	return http.postForm("/gallery/uploadImage/"+galleryId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new GalleryService();

File: src/main/java/com/myimages/service/GalleryService.java
package com.myimages.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.myimages.domain.Gallery;
import com.myimages.dto.GalleryDTO;
import com.myimages.dto.GallerySearchDTO;
import com.myimages.dto.GalleryPageDTO;
import com.myimages.dto.GalleryConvertCriteriaDTO;
import com.myimages.service.GenericService;
import com.myimages.dto.common.RequestDTO;
import com.myimages.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface GalleryService extends GenericService<Gallery, Integer> {

	List<Gallery> findAll();

	ResultDTO addGallery(GalleryDTO galleryDTO, RequestDTO requestDTO);

	ResultDTO updateGallery(GalleryDTO galleryDTO, RequestDTO requestDTO);

    Page<Gallery> getAllGallerys(Pageable pageable);

    Page<Gallery> getAllGallerys(Specification<Gallery> spec, Pageable pageable);

	ResponseEntity<GalleryPageDTO> getGallerys(GallerySearchDTO gallerySearchDTO);
	
	List<GalleryDTO> convertGallerysToGalleryDTOs(List<Gallery> gallerys, GalleryConvertCriteriaDTO convertCriteria);

	GalleryDTO getGalleryDTOById(Integer galleryId);


}
